cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -248,6 +249,23 @@ class UninitializedNode extends Node, TUninitializedNode {
   LocalVariable getLocalVariable() { result = v }
 }
 
+/** INTERNAL: do not use. The final value of a non-const ref parameter. */
+class RefParameterFinalValueNode extends Node, TRefParameterFinalValueNode {
```

Comment:
Could this go in `DataFlowPrivate.qll`?

Reply:
Not without some restructuring. The name TRefParameterFinalValueNode is not
available outside this file.